# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/gamepad/GamepadManager.h
# Commit: 8132dd939023
# Full Hash: 8132dd9390233328c8da7072f36859424a2b079d
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-10-14 21:42:48
# Regressor Bug: 1657404
# File Overlap Count: 2
# Description:
#   Bug 1657404 - Change PGamepadEventChannel to "refcounted protocol" r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D93119
# ==============================================================================

diff -r 186ca9e93e1d -r 8132dd939023 dom/gamepad/GamepadManager.h
--- a/dom/gamepad/GamepadManager.h	Wed Oct 07 11:34:02 2020 +0000
+++ b/dom/gamepad/GamepadManager.h	Wed Oct 14 15:58:36 2020 +0000
@@ -119,11 +119,7 @@
   // true when shutdown has begun
   bool mShuttingDown;
 
-  // Gamepad IPDL child
-  // This pointer is only used by this singleton instance and
-  // will be destroyed during the IPDL shutdown chain, so we
-  // don't need to refcount it here.
-  nsTArray<GamepadEventChannelChild*> mChannelChildren;
+  nsTArray<RefPtr<GamepadEventChannelChild>> mChannelChildren;
 
  private:
   nsresult Init();